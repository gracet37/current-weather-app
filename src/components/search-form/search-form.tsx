import React from "react";
import { TextField } from '@mui/material';

type SearchFormProps = {
    handleInputChange: (input: string ) => void;
}

export const SearchForm: React.FC<SearchFormProps> = ({ handleInputChange }) => {

    return (
        <React.Fragment>
            <TextField 
                aria-label="city search input" 
                fullWidth
                id="outlined-basic" 
                label="City search" 
                variant="outlined" 
                onChange={(e) => handleInputChange(e.target.value)}
            />
        </React.Fragment>
    )
};

