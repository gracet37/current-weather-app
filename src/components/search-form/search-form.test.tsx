import React from 'react';
import renderer from 'react-test-renderer';
import { SearchForm } from './search-form';

it('renders correctly', () => {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    const handleInputChange = () => {};
    const tree = renderer
        .create(<SearchForm handleInputChange={handleInputChange} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
