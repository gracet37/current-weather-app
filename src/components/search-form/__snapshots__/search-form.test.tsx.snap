// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  aria-label="city search input"
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor="outlined-basic"
    id="outlined-basic-label"
  >
    City name
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="outlined-basic"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-173wfxe"
      >
        <span>
          City name
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
